operator/pkg/clustermesh/clustermesh.go

```diff
@@ -0,0 +1,147 @@
+	common common.ClusterMesh
+
+	Metrics Metrics
+	// globalServices is a list of all global services. The datastructure
+	// is protected by its own mutex inside the structure.
+	globalServices *globalServiceCache
+	stopCh         chan struct{}
+
+	storeFactory store.Factory
+
+	conurrentClusterMeshEndpointSync int
+
+	meshPodInformer     *meshPodInformer
+	meshNodeInformer    *meshNodeInformer
+	meshServiceInformer *meshServiceInformer
+
+	endpointSliceMeshController  *endpointslice.Controller
+	endpointSliceInformerFactory informers.SharedInformerFactory
+}
+
+func newClusterMesh(lc hive.Lifecycle, params clusterMeshParams) *clusterMesh {
+	if !params.Clientset.IsEnabled() || params.ClusterMeshConfig == "" {
+		return nil
+	}
+
+	cm := clusterMesh{
+		Metrics: params.Metrics,
+		globalServices: newGlobalServiceCache(
+			params.Metrics.TotalGlobalServices.WithLabelValues(params.ClusterInfo.Name),
+		),
+		stopCh:                           make(chan struct{}),
+		storeFactory:                     params.StoreFactory,
+		conurrentClusterMeshEndpointSync: params.Cfg.ConurrentClusterMeshEndpointSync,
+	}
+	cm.meshPodInformer = &meshPodInformer{GlobalServiceCache: cm.globalServices}
+	cm.meshServiceInformer = newMeshServiceInformer(cm.globalServices, params.Clientset)
+	cm.meshNodeInformer = newMeshNodeInformer()
+	cm.endpointSliceMeshController, cm.endpointSliceInformerFactory = newEndpointSliceMeshController(
+		params.Cfg,
+		cm.meshPodInformer,
+		cm.meshServiceInformer,
+		cm.meshNodeInformer,
+		params.Clientset,
+	)
+	cm.common = common.NewClusterMesh(common.Configuration{
+		Config:           params.Config,
+		ClusterInfo:      params.ClusterInfo,
+		NewRemoteCluster: cm.newRemoteCluster,
+		Metrics:          params.CommonMetrics,
+	})
+
+	lc.Append(&cm.common)
+	lc.Append(&cm)
+	return &cm
+}
+
+func (cm *clusterMesh) newRemoteCluster(name string, status common.StatusFunc) common.RemoteCluster {
+	rc := &remoteCluster{
+		name:         name,
+		mesh:         cm,
+		storeFactory: cm.storeFactory,
+		synced:       newSynced(),
+	}
+
+	rc.remoteServices = cm.storeFactory.NewWatchStore(
+		name,
+		func() store.Key { return new(serviceStore.ClusterService) },
+		&remoteServiceObserver{remoteCluster: rc},
+		store.RWSWithOnSyncCallback(func(ctx context.Context) { close(rc.synced.services) }),
+	)
+
+	return rc
+}
+
+func (cm *clusterMesh) Start(hive.HookContext) error {
+	// Run EndpointSlice sync
+	cm.endpointSliceInformerFactory.Start(cm.stopCh)
+	go cm.meshServiceInformer.Run(cm.stopCh)
+	cm.endpointSliceInformerFactory.WaitForCacheSync(cm.stopCh)
+
+	if !cache.WaitForCacheSync(cm.stopCh, cm.meshServiceInformer.HasSynced) {
+		return fmt.Errorf("WaitForCacheSync on service informer not successful")
+	}
+
+	go cm.endpointSliceMeshController.Run(context.Background(), cm.conurrentClusterMeshEndpointSync)
+
+	return nil
+}
+
+func (cm *clusterMesh) Stop(hive.HookContext) error {
+	close(cm.stopCh)
+	return nil
+}
```

Comment:
To do that we would need to patch this: https://github.com/cilium/endpointslice-controller/blob/main/endpointslice/endpointslice_controller.go#L299 shouldn't be too hard but that's still something that would need to be maintained.

Not sure if it's worthwhile, happy to do it if you think that it is though!

Reply:
I guess it could make sense to wait for the ones spawned by the clustermesh cell. I agree that there's little point in patching the upstream fork of the library just for this.